libs/documentation/src/lib/components/slide-out/demos/basic/slide-out-basic.component.ts

```diff
@@ -0,0 +1,32 @@
+import { ScrollStrategyOptions } from '@angular/cdk/overlay';
+import { Component } from '@angular/core';
+import { SdsDialogService } from '@gsa-sam/components';
+
+@Component({
+  templateUrl: './slide-out-basic.component.html',
+  styleUrls: ['./slide-out-basic.component.scss'],
+  selector: `sds-slide-out-basic-demo`
+})
+export class SlideOutBasicComponent {
+
+  constructor(public dialog: SdsDialogService, private options: ScrollStrategyOptions) {}
+
+  buttonClicked(){
+    this.dialog.open(SlideOutTemplateComponent, {
```

Comment:
Could you update this and the template-ref demo to have toggle behavior rather than open on click - maybe keep reference of the dialog, and if the button is clicked, and the dialog is currently open, we close rather than open another one.